AdblockPlusSafariExtension/AdblockPlus+Parsing.m

```diff
@@ -0,0 +1,308 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-2015 Eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/&gt.
+ */
+
+#import "AdblockPlus+Parsing.h"
+
+// yajl is sax-like json parser. Content blocker extension has limited amount of memory,
+// so that it is not possible to load whole filter list at once.
+#include <yajl/yajl_parse.h>
+#include <yajl/yajl_gen.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct
+{
+  NSUInteger array_level;
+  yajl_gen g;
+
+} adblock_plus_context;
+
+static int reformat_null(void *ctx)
+{
+  adblock_plus_context *context = (adblock_plus_context *)ctx;
+  return yajl_gen_status_ok == yajl_gen_null(context->g);
+}
+
+static int reformat_boolean(void *ctx, int boolean)
+{
+  adblock_plus_context *context = (adblock_plus_context *)ctx;
+  return yajl_gen_status_ok == yajl_gen_bool(context->g, boolean);
+}
+
+static int reformat_number(void *ctx, const char *s, size_t l)
+{
+  adblock_plus_context *context = (adblock_plus_context *)ctx;
+  return yajl_gen_status_ok == yajl_gen_number(context->g, s, l);
+}
+
+static int reformat_string(void *ctx, const unsigned char * stringVal, size_t stringLen)
```

Comment:
Nit: Snuggle * to the variable name, i.e. `*stringVal` for consistency? Same below.